core/src/hlist.rs

```diff
@@ -183,11 +213,18 @@ pub struct HCons<H, T> {
     pub tail: T,
 }
 
+#[cfg(feature = "typenum")]
+impl<H, T: HList> HList for HCons<H, T>
+where
+    <T as HList>::Len: Add<B1>,
+    <<T as HList>::Len as Add<B1>>::Output: Unsigned,
+{
+    type Len = <<T as HList>::Len as Add<B1>>::Output;
+    const LEN: usize = 1 + <T as HList>::LEN;
+}
+#[cfg(not(feature = "typenum"))]
 impl<H, T: HList> HList for HCons<H, T> {
     const LEN: usize = 1 + <T as HList>::LEN;
```

Comment:
Any suggestions RE: this duplication? it's exactly the same except for the added where-clases, but if there's a way to feature gate where-clases only, I'm unfamiliar